fedimint-client/src/sm/state.rs

```diff
@@ -84,7 +84,7 @@ type TriggerFuture = Pin<Box<dyn Future<Output = serde_json::Value> + Send + 'st
 // TODO: remove Arc, maybe make it a fn pointer?
 type StateTransitionFunction<S> = Arc<
     dyn for<'a> Fn(
-            &'a mut IsolatedDatabaseTransaction<'_, '_, ModuleInstanceId>,
+            &'a mut ModuleDatabaseTransaction<'_, ModuleInstanceId>,
```

Comment:
nit: `ModuleDatabaseTrabsaction` could be a type def instead:

```rust
type ModuleDatabaseTransaction<'a, 'b> = IsolatedDatabaseTransaction<'a, 'b, ModuleInstanceId>;
```

Reply:
That might be a little bit leaky though.